### PR TITLE
Adds a catch-all error page to display uncaught errors.

### LINKED ISSUE #883: P1: Improve catch all error page
This error page is the outermost catch all on an application error.
The current error screen is created via `PrimeErrorBoundary` in `app.tsx` and `patientApp.tsx` The error dumps a json blob of the error.

- [ ] Update design of the error page to: 

- [ ] send the JSON error to Azure Application Insights

### PR DESCRIPTION
**Pull request by emmastephenson**
Adds the new page and styling, and updates the uncaught error message across the app.

## Related Issue or Background Info

- Fixes #883 

## Changes Proposed

- Add a new error page
- updates error shown in PatientApp.tsx and App.tsx

## Additional Information

I believe the App Insights tracking has already been implemented in PrimeErrorBoundary.

## Screenshots / Demos
Screen Shot 2021-03-31 at 12 36 25 PM

### CODE REVIEW

**Comment by emmastephenson** [comment]
SonarCloud is failing with insufficient test coverage, but I'm not sure how to test that the error page is being thrown. The page only shows up if there's an uncaught exception on the backend, and I don't know of a way to mock that for either e2e or unit tests. Advice or recommendations appreciated!

**Comment by maya** [comment]
I changed the support@simplereport.gov text to a `mailto` link so the `ErrorPage.test.tsx` file will also need to be updated but I didn't know how to. 

I updated the design to:

localhost_3000_error_facility=39a46fee-8ebf-4277-81ff-dfeb7cebc7dc

**Review by maya — APPROVED** [review]
Front-end LGTM. 

Left a comment to update the test file.

File: frontend/src/app/PrimeErrorBoundary.js [hunk 1]
```diff
@@ -18,10 +19,7 @@ export default class PrimeErrorBoundary extends React.Component {
 
   render() {
     if (this.state.hasError) {
-      const { onError } = this.props;
-      return typeof onError === "function"
-        ? onError(this.state.error)
-        : React.createElement(onError);
+      return React.createElement(ErrorPage);
```
Comment (ClydeDroid) [inline frontend/src/app/PrimeErrorBoundary.js:22]: Nice, no need to pass it as a prop :+1:

File: frontend/src/app/PrimeErrorBoundary.test.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    appInsights: { trackException: jest.fn() },
+  };
+});
+
+class ComponentOfDeath extends Component<any, any> {
+  public componentDidMount() {
+    throw new Error("a really terrible uncaught exception!");
+  }
+  public render() {
+    return <h1>Component of doooom</h1>;
```
Comment (ClydeDroid) [inline frontend/src/app/PrimeErrorBoundary.test.tsx:18]: :joy:

File: frontend/src/app/PrimeErrorBoundary.test.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+  it("renders the error page on component error", () => {
+    const { container } = render(
+      <MemoryRouter>
+        <PrimeErrorBoundary>
+          <ComponentOfDeath></ComponentOfDeath>
+        </PrimeErrorBoundary>
+      </MemoryRouter>
+    );
+    expect(container).toHaveTextContent("Something went wrong :(");
```
Comment (ClydeDroid) [inline frontend/src/app/PrimeErrorBoundary.test.tsx:39]: Great test!!

**Review by ClydeDroid — APPROVED** [review]
Awesome work!!!

File: frontend/src/app/PrimeErrorBoundary.js (hunk 1 above)
Comment (emmastephenson) [reply frontend/src/app/PrimeErrorBoundary.js:22]: I wrote it this way to get around writing specific error tests for App and PatientApp, but let's call it a feature not a bug 😂

File: frontend/src/app/PrimeErrorBoundary.test.tsx (hunk 3 above)
Comment (emmastephenson) [reply frontend/src/app/PrimeErrorBoundary.test.tsx:39]: Thanks! And thanks for the resources you sent me, they helped a lot.

**Pull request merged**